src/model_loader.hpp

```diff
@@ -0,0 +1,187 @@
+#pragma once
+#include <vector>
+
+namespace wr
+{
+	struct ModelMeshData;
+	struct ModelMaterialData;
+	struct ModelData;
+	struct ModelSkeletonData;
+	struct ModelBoneData;
+	struct ModelAnimationData;
+
+	struct EmbeddedTexture;
+
+	enum class TextureLocation
+	{
+		EXTERNAL = 0,
+		EMBEDDED,
+		NON_EXISTENT
+	};
+
+	struct Vector3
```

Comment:
Why the need for custom Vector3/4 and Matrix classes? And if they can't be removed, they should at least be hidden. Putting them in the wr namespace and in an hpp file means that you're exposing them to the user and could be used outside your original intentions.

Reply:
I need to access this data in the model pool, and I can't use the DX12 versions.